src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/Query/FilterFacetsParser.cs

```diff
@@ -0,0 +1,155 @@
+
+internal class FilterFacetsParser : IFacetsParser
+{
+    private readonly QueryState _queryState;
+    private readonly Dictionary<string, ParquetFilterOption> _filterOptionsByPublicId;
+
+    public FilterFacetsParser(QueryState queryState, IEnumerable<ParquetFilterOption> filterOptions)
+    {
+        _queryState = queryState;
+        _filterOptionsByPublicId = filterOptions.ToDictionary(o => o.PublicId);
+    }
+
+    public IInterpolatedSql Parse(DataSetQueryCriteriaFacets facets, string path)
+    {
+        var fragments = new List<IInterpolatedSql>();
+
+        if (facets.Filters?.Eq is not null)
+        {
+            fragments.Add(
+                EqFragment(
+                    filterOptionId: facets.Filters.Eq,
+                    path: QueryUtils.Path(path, "filters.eq")
+                )
+            );
+        }
+
+        if (facets.Filters?.NotEq is not null)
+        {
+            fragments.Add(
+                EqFragment(
+                    filterOptionId: facets.Filters.NotEq,
+                    path: QueryUtils.Path(path, "filters.notEq"),
+                    negate: true
+                )
+            );
+        }
+
+        if (facets.Filters?.In is not null && facets.Filters.In.Count != 0)
+        {
+            fragments.Add(
+                InFragment(
+                    filterOptionIds: [..facets.Filters.In],
+                    path: QueryUtils.Path(path, "filters.in")
+                )
+            );
+        }
+
+        if (facets.Filters?.NotIn is not null && facets.Filters.NotIn.Count != 0)
+        {
+            fragments.Add(
+                InFragment(
+                    filterOptionIds: [..facets.Filters.NotIn],
+                    path: QueryUtils.Path(path, "filters.notIn"),
+                    negate: true
+                )
+            );
+        }
+
+        return new DuckDbSqlBuilder()
+            .AppendRange(fragments, "\nAND ")
+            .Build();
+    }
+
+    private IInterpolatedSql EqFragment(
+        string filterOptionId,
+        string path,
+        bool negate = false)
+    {
+        var builder = new DuckDbSqlBuilder();
+
+        if (!_filterOptionsByPublicId.TryGetValue(filterOptionId, out var option))
+        {
+            _queryState.Warnings.Add(CreateNotFoundWarning([filterOptionId], path));
+
+            return builder
+                .AppendLiteral(negate ? "true" : "false")
+                .Build();
+        }
+
+        builder += $"{DataTable.Ref().Col(option.ColumnName):raw} {(negate ? "!=" : "="):raw} {option.Id}";
+
+        return builder.Build();
+    }
+
+    private IInterpolatedSql InFragment(
+        HashSet<string> filterOptionIds,
+        string path,
+        bool negate = false)
+    {
+        var builder = new DuckDbSqlBuilder();
+
+        var options = filterOptionIds
+            .Distinct()
+            .Select(id => _filterOptionsByPublicId.GetValueOrDefault(id))
+            .OfType<ParquetFilterOption>()
```

Comment:
If we did this, it'd be likely for exceptions to thrown as the `id` may not actually exist as a key of `_filterOptionsByPublicId`.

We do it the current way so that we get nulls if the key doesn't exist, and then we filter out the nulls using `OfType`.